File: client-scalaz/src/main/scala/org/openapitools/client/api/PreferredLocationResponse.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime


import PreferredLocationResponse._

case class PreferredLocationResponse (
  ownerId: Option[Long],
onDate: Option[String],
onDateMilli: Option[Long],
hour: Option[Integer],
latitude: Option[Double],
longitude: Option[Double],
distance: Option[Double],
duration: Option[Long],
startStepId: Option[Long],
endStepId: Option[Long])

object PreferredLocationResponse {
  import DateTimeCodecs._

  implicit val PreferredLocationResponseCodecJson: CodecJson[PreferredLocationResponse] = CodecJson.derive[PreferredLocationResponse]
  implicit val PreferredLocationResponseDecoder: EntityDecoder[PreferredLocationResponse] = jsonOf[PreferredLocationResponse]
  implicit val PreferredLocationResponseEncoder: EntityEncoder[PreferredLocationResponse] = jsonEncoderOf[PreferredLocationResponse]
}

File: client-clojure/src/sirqul_io_t_platform/specs/preferred_location_response.clj
(ns sirqul-io-t-platform.specs.preferred-location-response
  (:require [clojure.spec.alpha :as s]
            [spec-tools.data-spec :as ds]
            )
  (:import (java.io File)))


(def preferred-location-response-data
  {
   (ds/opt :ownerId) int?
   (ds/opt :onDate) string?
   (ds/opt :onDateMilli) int?
   (ds/opt :hour) int?
   (ds/opt :latitude) float?
   (ds/opt :longitude) float?
   (ds/opt :distance) float?
   (ds/opt :duration) int?
   (ds/opt :startStepId) int?
   (ds/opt :endStepId) int?
   })

(def preferred-location-response-spec
  (ds/spec
    {:name ::preferred-location-response
     :spec preferred-location-response-data}))
